### DISCUSSION TITLE
[DOCS]: getShopPaymentAccountLedgerEntries: enum for ledger_type, reference_type

### DISCUSSION THREAD

**Discussion by corsoa** [etsy/open-api#738] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Link to Documentation Issue

https://developer.etsy.com/documentation/reference#operation/getShopPaymentAccountLedgerEntries

### What would you like to change?

I would like for the specific ledger_type entries to be an enum, similar to other response entities (see example: https://developer.etsy.com/documentation/reference#operation/getShopPaymentAccountLedgerEntries response schema `state` property).

Specific values I'd like added: `offsite_ads_fee`, `transaction`, `shipping_transaction`, `gift_wrap_fees`, `renew_sold_auto`, `listing`, `renew_sold`, `listing_private`, `auto_renew_expired`, `transaction_quantity`, `renew_expired`, `REFUND`, `transaction_quantity_refund`, `transaction_refund`, `listing_refund`, `shipping_transaction_refund`, `offsite_ads_fee_refund`, `renew_sold_auto_refund`, `listing_private_refund`, `shipping_label_refund`, `DISBURSE2`, `shipping_labels`, and `sales_tax`. (+ any others that are possible but I don't have)

I would also like `reference_type` to be an enum to describe the linked object based on the `ledger_type`.

### Why are you proposing this change?

This would allow developers to be able to more easily see the potential types of ledgers and referenced objects.

For example - `offsite_ads_fee_refund` ledger type includes a non-nullable int for the reference_id, but the `reference_type` is 'etsy'. It is not clear what type of object this is; I can only infer the ID is not a receipt and not a ledger. (it was mentioned on this issue: https://github.com/etsy/open-api/issues/225 that `offsite_ads_fee` is a receipt, but this reference ID for the refund is not the receipt ID)

**Comment by etsyachristensen** [comment]
We're moving this to the Idea discussion because we are closing the issues board soon.

**Reply by agrossjohann** [reply]
Hey Andrew.  Just wondering if there's any documentation available per @corsoa 's post?  We're working on a bookkeeping app and don't want to miss (or miscalculate) any ledger entries.  A complete list of ledger_types and explanations of how these entries hit the accounts would be extremely helpful.  Thank you!

**Comment by agrossjohann** [comment]
Thanks for requesting this.  We're also interested in having a complete list of ledger_type strings, as well as in knowing how they map in reference to the financial CSV download (sale, refund, fee, etc...).
